### ISSUE TITLE
problem with gcc trunck (gcc 15.0.0)

### ISSUE THREAD

**Issue by blackgolfer** [Tencent/rapidjson#2277]
rapidjson/document.h:319:82: error: assignment of read-only member ‘rapidjson::GenericStringRef<CharType>::length’

**Comment by BlackZork** [comment]
Same on current ArchLinux:

```build] /usr/include/rapidjson/document.h: In member function ‘rapidjson::GenericStringRef<CharType>& rapidjson::GenericStringRef<CharType>::operator=(const rapidjson::GenericStringRef<CharType>&)’:
[build] /usr/include/rapidjson/document.h:319:82: error: assignment of read-only member ‘rapidjson::GenericStringRef<CharType>::length’
[build]   319 |     GenericStringRef& operator=(const GenericStringRef& rhs) { s = rhs.s; length = rhs.length; }
[build]       |                                                                           ~~~~~~~^~~~~~~~~~~~```

gcc version 14.1.1 20240507 (GCC)

**Comment by l1nxy** [comment]
Need add patch from https://github.com/Tencent/rapidjson/pull/719 .

**Comment by blackgolfer** [comment]
yes, it is the problem with current release v1.1.0 which seems to be out of date.

**Comment by luis-grijalva** [comment]
I'm still seeing this error while trying to build osmium-tool on Arch Linux, is there an update? I'm using version 1.1.0-6 and GCC 14.1.1

**Comment by lahwaacz** [comment]
The fix (#719) was merged in 2016 :scream: 

When can we expect a new release that would include these changes?

**Comment by amlamarra** [comment]
> The fix (#719) was merged in 2016 😱
> 
> When can we expect a new release that would include these changes?

Seconded. Still waiting for a new release!

**Comment by fgalan** [comment]
> > The fix ([#719](https://github.com/Tencent/rapidjson/pull/719)) was merged in 2016 😱
> > When can we expect a new release that would include these changes?
> 
> Seconded. Still waiting for a new release!

Is somebody still maintaining rapidjson?

No new release since 1.1.0 in 10 years... :/

**Comment by ic** [comment]
Work around that unlocked my builds: https://github.com/BYVoid/OpenCC/commit/e1f3e6b17ebd70c799926e675fe8d1ec72f6b36a

Quick scan of the code and it looks like this is spot on: The choke point looks like unused code (quick grep, not fully verified) and just fine to comment out.

**Comment by cscd98** [comment]
Still seeing this with GCC 15 now.
